CMakeLists.txt

```diff
@@ -80,7 +81,7 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
 endif()
 add_library(awkward-static STATIC $<TARGET_OBJECTS:awkward-objects>)
 set_property(TARGET awkward-static PROPERTY POSITION_INDEPENDENT_CODE ON)
-add_library(awkward        SHARED $<TARGET_OBJECTS:awkward-objects>)
+add_library(awkward        SHARED $<TARGET_OBJECTS:awkward-objects> include/awkward/kernels/kernel.h src/kernels/kernel.cpp src/gpu-kernels/identities.cpp include/awkward/gpu-kernels/identities.h)
```

Comment:
I don't know what's going on here, but the gpu-kernel sources should be identified by a glob pattern and passed around the CMake configuration as a variable.

Reply:
I'm not sure what's happening in this CMake, but be sure that the new dispatching functions are in libawkward.so, not one of the libawkward-cpu-kernels.so or libawkward-gpu-kernels.so dependencies.

Reply:
Yeah, I fixed this in the latest commit. 
One thing that has been bugging me is that, in the Linux build, somehow the `libawkward_gpu_kernels.so` can access and link with `libawkward_cpu_kernels.so`, which shouldn't be the case. I stumbled upon this when I saw the mac and windows build failing. It was because in the `gpu_kernels` files I was accessing `cpu_kernels/util.h`. I have no idea why Linux builds passed. Have you ever encountered this?